client/layout/masterbar/oauth-client.jsx

```diff
@@ -23,10 +28,16 @@ const DefaultOauthClientMasterbar = ( { oauth2Client } ) => (
 		<nav>
 			<ul className="masterbar__oauth-client-main-nav">
 				<li className="masterbar__oauth-client-current">
-					{ oauth2Client.icon && (
+					{ isJetpackCloudOAuth2Client( oauth2Client ) ? (
```

Comment:
Instead of 

```
{ isJetpackCloudOAuth2Client( oauth2Client ) ? (
						<div className="masterbar__oauth-client-logo">
							<JetpackLogo full monochrome={ false } size={ 28 } />
						</div>
					) : (
						oauth2Client.icon && (
							<div className="masterbar__oauth-client-logo">
								<img src={ oauth2Client.icon } alt={ oauth2Client.title } />
							</div>
						)
					) }
```

can we define the 
```
const oauthLogo = this.getOAuthLogo( oauth2Client ); 
```

Would be a new method that retuns null or the image or Jetpack logo based on the oauth2Client that is passed in. Then we can simply things with 
```
{ oauthLogo && <div className="masterbar__oauth-client-logo">oauthLogo</div>}
```

Reply:
Agree with @enejb.